swift/Arrow/Sources/Arrow/ArrowArray.swift

```diff
@@ -233,3 +273,58 @@ public class BinaryArray: ArrowArray<Data> {
         }
     }
 }
+
+public class StructArray: ArrowArray<[Any?]> {
+    public private(set) var arrowFields: [ArrowArrayHolder]?
+    public required init(_ arrowData: ArrowData) {
+        super.init(arrowData)
+    }
+
+    public func initialize() throws -> StructArray {
```

Comment:
I'm not familiar with Swift but why do we need to define `initialize()`?
Can we do this in `init()`? (`init()` should not throw?)

Reply:
Correct, ArrowArray's init does not throw.  The ArrowArrayHolderImpl.loadArray method can throw so in order to allow this to be thrown I added the initialize method.  (I looked at adding a convenience init method but the throws keyword isn't considered in overloading.)